File: LeetCode/cpp/1844_Replace_All_Digits_with_Characters.cpp
class Solution {
public:
	string replaceDigits(string s) {
		string answer;

		for (int i = 0; i < s.size(); i++) {
			if (i % 2 == 0) {
				answer += s[i];
				continue;
			}

			char c = ((s[i - 1] - 'a') + (s[i] - '0')) + 'a';
			answer += c;
		}

		return answer;
	}
};

File: LeetCode/java/1844_Replace_All_Digits_with_Characters.java
class Solution {
    public String replaceDigits(String s) {
        StringBuilder answer = new StringBuilder();

        for (int i = 0; i < s.length(); i++) {
            if (i % 2 == 0) {
                answer.append(s.charAt(i));
                continue;
            }

            int alphabet = s.charAt(i - 1) - 'a';
            int number = s.charAt(i) - '0';
            char c = (char)((alphabet + number) + 'a');

            answer.append(c);
        }

        return answer.toString();
    }
}